docs/manual/technical_design.md

```diff
@@ -331,6 +333,58 @@ The restore reconciler runs the following subreconcilers:
+5. The operator tells the sidecar to regenerate the monitor conf based on the new template. The sidecar places the generated monitor conf in its output directory.
+6. The operator checks the latest output monitor conf to see if it is now correct.
+7. Once the monitor conf is correct, the operator uses the CLI to shut down the fdbserver processes.
+8. fdbmonitor sees that that processes have exited and starts new processes with the latest configuration.
+
+The operator follows a similar process when the `fdb.cluster` file needs to be updated. However, because this file is not templated, the sidecar simply copies the file from the input directory to the output directory. Cluster file updates do not require restarting processes.
+
+When the operator checks the status of the cluster, it needs to check if the process start commands are an exact match for the expected values based on the cluster spec. In order to make this comparison, it needs to fill in pod-specific information like the address and node name. The sidecar also provides an API for reading the environment variables that are being referenced in the monitor conf, and what their current values are. The operator uses this API when performing this check on the start command.
+
+The sidecar has an important role to play in the upgrade flow. The monitor conf template uses a template variable `$BINARY_DIR` for the directory where the `foundationdb` container should look for the `fdbserver` binary. The sidecar process sets this template variable based on its understanding of the versions of the main container and the sidecar container. When they are running the same version of FDB, the `$BINARY_DIR` is set to the directory with the binaries that are provided by the `foundationdb` image. When they are running a different version, the sidecar copies the FDB binaries from its own image into the output directory, and sets the `$BINARY_DIR` to the path to these binaries in that directory.
+
+### Unified Image
+
+**NOTE**: The unified image is still experimental, and is not recommended outside of development environments.
+
+When using the unified image, the `foundationdb` container runs a `fdb-kubernetes-monitor` process, which is responsible for starting `fdbserver` processes. `fdb-kubernetes-monitor` receives its configuration in the form of a JSON file, which provides the command-line arguments in a structured form. These arguments can reference environment variables, which will be filled in by `fdb-kubernetes-monitor`. They can also reference the process number, which allows `fdb-kubernetes-monitor` to start multiple `fdbserver` processes that use different ports and different data directories.
+
+The flow for updating the monitor conf file has the following steps:
+
+1. The operator updates the monitor conf template in the `sample-cluster-config` config map. There is one monitor conf file for every process class the cluster uses.
```

Comment:
```suggestion
1. The operator updates the monitor conf template in the config map. There is one monitor conf file for every process class the cluster uses.
```

Reply:
Is there a problem with including the name of the config map? I thought it would be helpful to be more specific about the config map we're discussing.